src/test/java/api/requests/RequestsAPILoanRenewalTests.java

```diff
@@ -254,4 +436,61 @@ public void multipleRenewalFailuresWhenItemHasOpenRecallRequestAndLoanIsNotLoana
     assertThat(response.getJson(), hasErrorWith(hasMessage(ITEMS_CANNOT_BE_RENEWED_MSG)));
     assertThat(response.getJson(), hasErrorWith(hasMessage(EXPECTED_REASON_LOAN_IS_NOT_LOANABLE)));
   }
+
+  private void loanPolicyWithRollingProfileAndRenewingIsForbiddenWhenHoldIsPending()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    LoanPolicyBuilder dueDateLimitedPolicy = new LoanPolicyBuilder()
+      .withName("Rolling. Renewing is forbidden with holds requests")
+      .withDescription("Rolling profile with disallowed renewing a loan when a hold request is pending")
+      .rolling(Period.weeks(3))
+      .unlimitedRenewals()
+      .renewFromSystemDate();
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.create(dueDateLimitedPolicy).getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.activeNotice().getId()
+    );
+  }
+
+  private void loanPolicyWithFixedProfileAndRenewingIsForbiddenWhenHoldIsPending()
+    throws InterruptedException, MalformedURLException, TimeoutException, ExecutionException {
+    LocalDate now = LocalDate.now();
+    FixedDueDateSchedulesBuilder fixedDueDateSchedules = new FixedDueDateSchedulesBuilder()
+      .withName("1 month - Fixed Due Date Schedule")
+      .addSchedule(wholeMonth(now.getYear(), now.getMonthValue()));
+
+    LoanPolicyBuilder dueDateLimitedPolicy = new LoanPolicyBuilder()
+      .withName("Fixed Due Date Policy")
+      .fixed(loanPoliciesFixture.createSchedule(fixedDueDateSchedules).getId())
+      .renewFromSystemDate();
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.create(dueDateLimitedPolicy).getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.activeNotice().getId()
+    );
+  }
+
+  private void useRollingPolicyWithRenewingAllowedForHoldingRequest()
+    throws InterruptedException, MalformedURLException, TimeoutException, ExecutionException {
+    JsonObject holds = new JsonObject();
+    holds.put("alternateRenewalLoanPeriod", Period.weeks(3).asJson());
```

Comment:
I would prefer an **alternate renewal period** to be different from the **loan period** for these tests to verify we use correct period.

Reply:
done.